src/driver/interface/maildriver.c

```diff
@@ -52,7 +52,9 @@ mailsession * mailsession_new(mailsession_driver * sess_driver)
   int r;
   
   session = malloc(sizeof(* session));
-  
+  if (session == NULL)
+    return NULL;
```

Comment:
could you instead use:
`goto err;`
and add a `err:` label at the bottom before `return NULL`?

Reply:
That's the pattern that's generally used in the code for error handling & proper clean up.

Reply:
Sure. Done.